ecommerce-ios/Views/SizeSelection/SizeSelectionView.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  SizeSelectionView.swift
+//  ecommerce-ios
+//
+//  Created by Nguyen M. Tam on 19/05/2021.
+//
+
+import SwiftUI
+
+struct SizeSelectionView: View {
+
+    @State private var selectedViewModel: SizeCellViewModel
+
+    let cellViewModels: [SizeCellViewModel]
+
+    private let spacing: CGFloat = 8.0
+    private let horizontalSpacing: CGFloat = 34.0
+
+    private var itemWidth: CGFloat {
+        let numberOfColumns = CGFloat(self.cellViewModels.count)
+        return ((screenWidth - spacing * (numberOfColumns + 1) - horizontalSpacing) / numberOfColumns).rounded(.down)
+    }
+
+    private var columns: [GridItem] {
+        [.init(.adaptive(minimum: itemWidth))]
+    }
```

Comment:
you can also
```

    private var columns: [GridItem] {
        .init(repeating: GridItem(.flexible()), count: self.cellViewModels.count)
    }
```
and
```
SizeCell(viewModel: cellViewModel)
                                .frame(height: 80.0)
```
making this a lot more reactive to parents view, what do you think?

Reply:
💯 💯 💯 💯